NetSSL_OpenSSL/include/Poco/Net/SecureSocketImpl.h

```diff
@@ -284,16 +284,21 @@ class NetSSL_API SecureSocketImpl
 		/// Callback to handle new session data sent by server.
 
 private:
+	using MutexT = Poco::FastMutex;
+	using LockT = MutexT::ScopedLock;
+	using UnLockT = Poco::ScopedLockWithUnlock<MutexT>;
+
 	SecureSocketImpl(const SecureSocketImpl&);
 	SecureSocketImpl& operator = (const SecureSocketImpl&);
 
-	SSL* _pSSL;
+	std::atomic<SSL*> _pSSL;
 	Poco::AutoPtr<SocketImpl> _pSocket;
 	Context::Ptr _pContext;
 	bool _needHandshake;
 	std::string _peerHostName;
 	Session::Ptr _pSession;
 	bool _bidirectShutdown = true;
+	mutable MutexT _mutex;
```

Comment:
Is `_mutex` intended to protect only _pSSL or also other related members (`_pContext`, `_pSession`)?

Reply:
mutex protects all openssl calls; `_pSSL` pointer was made atomic, smart pointers have atomic refcounting